src/annotator/anchoring/pdf.js

```diff
@@ -446,57 +446,71 @@ function prioritizePages(position) {
 /**
  * Anchor a set of selectors to a DOM Range.
  *
+ * `selectors` must include a `TextQuoteSelector` and may include other selector
+ * types.
+ *
  * @param {HTMLElement} root
- * @param {Selector[]} selectors - Selector objects to anchor
+ * @param {Selector[]} selectors
  * @return {Promise<Range>}
  */
-export function anchor(root, selectors) {
-  const position = /** @type {TextPositionSelector|undefined} */ (
-    selectors.find(s => s.type === 'TextPositionSelector')
-  );
+export async function anchor(root, selectors) {
   const quote = /** @type {TextQuoteSelector|undefined} */ (
     selectors.find(s => s.type === 'TextQuoteSelector')
   );
 
-  /** @type {Promise<Range>} */
-  let result = Promise.reject('unable to anchor');
+  // The quote selector is required in order to check that text position
+  // selector results are still valid.
+  if (!quote) {
+    throw new Error('No quote selector found');
+  }
+
+  const position = /** @type {TextPositionSelector|undefined} */ (
+    selectors.find(s => s.type === 'TextPositionSelector')
+  );
 
   if (position) {
-    result = result.catch(() => {
-      return findPage(position.start).then(({ index, offset, textContent }) => {
-        const start = position.start - offset;
-        const end = position.end - offset;
-        const length = end - start;
-
-        const matchedText = textContent.substr(start, length);
-        if (quote && quote.exact !== matchedText) {
-          throw new Error('quote mismatch');
-        }
+    // If we have a position selector, try using that first as it is the fastest
+    // anchoring method.
+    try {
+      const { index, offset, textContent } = await findPage(position.start);
+      const start = position.start - offset;
+      const end = position.end - offset;
+      const length = end - start;
+
+      const matchedText = textContent.substr(start, length);
+      if (quote.exact !== matchedText) {
+        throw new Error('quote mismatch');
+      }
 
-        return anchorByPosition(index, start, end);
-      });
-    });
-  }
+      const range = await anchorByPosition(index, start, end);
```

Comment:
```suggestion
      return anchorByPosition(index, start, end);
```

This matches the last return statement in this function.

Reply:
> This commit disallows anchoring using only position selectors in PDFs.

It seems that if `position` is available it returns the range by `anchorByPosition`. It doesn't seem to be any other condition check.